wordpress-mobile/WordPress-Android
# [Revamp Landing Screen] Jetpack Landing Screen design tweaks

Fixes #17214

Implements the latest design updates to the Jetpack Landing Screen. These are only UI tweaks to different elements, with no updates to the animation.

There's also a few code cleanup changes done to tidy up the Compose code. One particular change is that private composable methods lose their `modifier` parameter since these aren't to be called outside their single-use parent composable.

I've also changed the way the large texts are getting built and colored, making use of Compose's `AnnotatedString`. This was done to have a better control over the line height, since in the latest design update it was shrunk considerably, and by using a `Text` composable for each line of text it wasn't easy and clean to achieve the desired result.

To test:

<details>
<summary><b>Prerequisite</b> - Enable the <code>LandingScreenRevamp</code> feature flag</summary>

- Either `login` → `Me` → `App Settings` → `Debug Settings` → enable `LandingScreenRevamp` → `RESTART THE APP` → `logout` or
- Apply this hard-coded solution:

<details>
<summary>Patch to enable the <code>LANDING_SCREEN_REVAMP</code> build flag</summary>

```diff
diff --git a/WordPress/build.gradle b/WordPress/build.gradle
index 63260a4ca8..fe999fa92f 100644
--- a/WordPress/build.gradle
+++ b/WordPress/build.gradle
@@ -112,7 +112,7 @@ android {
         buildConfigField "boolean", "READER_COMMENTS_MODERATION", "false"
         buildConfigField "boolean", "SITE_INTENT_QUESTION", "true"
         buildConfigField "boolean", "SITE_NAME", "false"
-        buildConfigField "boolean", "LANDING_SCREEN_REVAMP", "false"
+        buildConfigField "boolean", "LANDING_SCREEN_REVAMP", "true"
         buildConfigField "boolean", "LAND_ON_THE_EDITOR", "false"
         buildConfigField "boolean", "BLOGGING_PROMPTS", "false"
         buildConfigField "boolean", "QUICK_START_EXISTING_USERS_V2", "false"
```  
</details>

---

</details>

##### 1️⃣ Test Case 1: Happy Flow

0. Open the JP app and make sure you're logged out
1. **Verify** That the _UI_ looks similar to the design specs:

<img width="679" alt="Screenshot 2022-09-29 at 13 50 10" src="https://user-images.githubusercontent.com/4588074/193023847-73c11189-545d-4a19-b7bb-406f7770f2e3.png">

2. Toggle dark mode on the device
3. **Verify** That the _UI on dark mode_ looks similar to the dark mode design specs.

## Regression Notes
1. Potential unintended areas of impact
   None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
   Manual testing, though no regression expected - the PR is adjusting the UI of a recently added screen that's behind a feature flag.

4. What automated tests I added (or what prevented me from doing so)
   N/a

## Previews

| Dark Mode | Light Mode |
| - | - |
| <img width="314" src="https://user-images.githubusercontent.com/4588074/193025210-beb0e66b-8e1e-4742-b7ba-2c7c03033a11.png"> | <img width="314" src="https://user-images.githubusercontent.com/4588074/193025187-b7422acf-7c04-4d3d-bd98-c5cf8e9f676f.png"> |

### 🎬 Video Preview
https://user-images.githubusercontent.com/4588074/193025599-2c254878-2ff2-452f-9a8a-adc4acf962d6.mp4

PR submission checklist:
